onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidateEmail.kt

```diff
@@ -0,0 +1,19 @@
+package com.alexafit.onboardingauthdomain.useCase
+
+class ValidateEmail {
+
+    operator fun invoke(email: String): Pair<String, Boolean> {
+        return when (email.isBlank()) {
+            true -> Pair(email.trim(), false)
+            false -> {
+                val trimmedEmail = email.trimEnd()
+                trimmedEmail.isEmailValid()
+            }
+        }
```

Comment:
I would personally prefer a simple if/else check if there are only 2 branches. Feels more readable to me